felix/bpf-apache/filter.h

```diff
@@ -12,26 +12,25 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include <linux/bpf.h>
 #include "bpf.h"
 
 struct protoport {
 	__u16 proto;
 	__u16 port;
 };
 
-struct bpf_map_def __attribute__((section("maps"))) calico_prefilter_v4 = {
-	.type           = BPF_MAP_TYPE_LPM_TRIE,
-	.key_size       = sizeof(union ip4_bpf_lpm_trie_key),
-	.value_size     = sizeof(__u32),
-	.max_entries    = 10240,
-	.map_flags      = BPF_F_NO_PREALLOC,
-};
+struct {
+    __uint(type, BPF_MAP_TYPE_LPM_TRIE);
+    __type(key, union ip4_bpf_lpm_trie_key);
+    __type(value, __u32);
+    __uint(max_entries, 10240);
+    __uint(map_flags, BPF_F_NO_PREALLOC);
+} calico_prefilter_v4 SEC(".map");
 
-struct bpf_map_def __attribute__((section("maps"))) calico_failsafe_ports = {
-	.type           = BPF_MAP_TYPE_HASH,
-	.key_size       = sizeof(struct protoport),
-	.value_size     = 1,
-	.max_entries    = 65535,
-	.map_flags      = BPF_F_NO_PREALLOC,
-};
+struct {
+    __uint(type, BPF_MAP_TYPE_HASH);
+    __type(key, struct protoport);
+    __type(value, __u32);
+    __uint(max_entries, 65535);
+    __uint(map_flags, BPF_F_NO_PREALLOC);
+} calico_failsafe_ports SEC(".map");
```

Comment:
is it correct to move it from `maps` to `.map` ? gpl maps as still in `.maps`

Reply:
yes, it should be `.maps`. I was troubleshooting this part, and your comment helped me to fix it. Thanks.